text/src/autogluon/text/automm/models/huggingface_text.py

```diff
@@ -14,12 +14,24 @@
     init_weights,
     get_column_features,
 )
-
+from .lora_layers import LoRALinear
 hf_logging.set_verbosity_error()
 
 logger = logging.getLogger(AUTOMM)
 
 
+def replace_layers_with_lora(model, old, lora_r, lora_alpha):
+    for n, module in model.named_children():
+        if len(list(module.children())) > 0:
+            replace_layers_with_lora(module, old, lora_r, lora_alpha)
+
+        if old in n:
+            lora_layer = LoRALinear(module.in_features, module.out_features, r= lora_r, lora_alpha=lora_alpha, merge_weights=False)
```

Comment:
We need to detect the type of module. Should we do the following instead? 

```
if isinstance(module, nn.Linear):
   lora_layer = LoRALinear(...)
```

Reply:
I agree that relying on the name is brittle. That said, LoRA is not applied to all linear layers, only query and value attention weights are replaced. I was thinking we might be able to maybe make a check on the type of parent module? But again here, different models define their own attention module (e.g. RobertaSelfAttention, DisentangledSelfAttention (Deberta)) so we might have to explicitly state which modules we support LoRA for?